main.go

```diff
@@ -71,59 +168,188 @@ func main() {
+		title = *pr.Title
+	}
 
-Title: %s
-Author: %s
-Description: %s
+	if pr.User != nil && pr.User.Login != nil {
+		author = *pr.User.Login
+	}
 
-Please provide a detailed code review that includes:
-1. Summary of what the PR does.
-2. Suggestions for improvements or refactoring.
-3. Potential bugs or issues to look out for.
-4. Any additional comments or concerns.
+	// Construct the full prompt with all file changes
+	var fileChanges []string
+	fileMap := make(map[string]*github.CommitFile)
+	for _, file := range files {
+		if file.Patch != nil {
+			fileChanges = append(fileChanges, fmt.Sprintf("File: %s\nPatch:\n%s", *file.Filename, *file.Patch))
+			fileMap[*file.Filename] = file
+		}
+	}
 
-Respond as a detailed and professional code review.
-`, *pr.Title, *pr.User.Login, *pr.Body)
+	combinedChanges := strings.Join(fileChanges, "\n\n")
+	prompt := fmt.Sprintf(`
+	PR %s by %s: %s
+	
+	The following files were changed:
+	%s
+	`, title, author, body, combinedChanges)
 
-	// Send the prompt to the assistant
 	resp, err := client.CreateChatCompletion(context.Background(), openai.ChatCompletionRequest{
-		Model: os.Getenv("ASSISTANT_MODEL"),
+		Model: openai.GPT4oMini,
 		Messages: []openai.ChatCompletionMessage{
 			{
 				Role:    openai.ChatMessageRoleUser,
 				Content: prompt,
 			},
 		},
+		User: os.Getenv("ASSISTANT_ID"),
 	})
 	if err != nil {
-		return "", err
+		return "", nil, "", err
 	}
 
-	// Return the generated review
-	return resp.Choices[0].Message.Content, nil
-}
+	responseText := resp.Choices[0].Message.Content
 
-// postReview posts a review on the PR with either approval or request for changes
-func postReview(client *github.Client, ctx context.Context, owner, repo string, prNumber int, review string, approve bool) error {
-	var state string
-	if approve {
-		state = "APPROVE"
+	// Parse the response to determine the action (approve or request changes)
+	var action string
+	if strings.Contains(strings.ToLower(responseText), "__approve__") {
+		action = "approve"
+	} else if strings.Contains(strings.ToLower(responseText), "__request_changes__") {
+		action = "request_changes"
 	} else {
-		state = "REQUEST_CHANGES"
+		action = "request_changes" // Default to requesting changes if unsure
 	}
 
+	// Example: Parse the response for line comments and overall feedback
+	// var reviewComments []*github.DraftReviewComment
+
+	// Mocked comment parsing logic; in reality, you would parse responseText
+	// mockComment := "This line could benefit from better error handling."
+
+	// for _, file := range files {
+	// 	if file.Patch != nil {
+	// 		// Ensure the file exists in the PR
+	// 		if fileMap[*file.Filename] {
+	// 			// Add a comment at a specific line, assuming line 12 for example
+	// 			reviewComments = append(reviewComments, &github.DraftReviewComment{
+	// 				Path:     file.Filename,
+	// 				Position: github.Int(12), // Example line number, should be valid
+	// 				Body:     github.String(mockComment),
+	// 			})
+	// 		}
```

Comment:
"This parsing logic appears fragile and may fail if the format changes. Consider using regular expressions or a more robust parser to extract comments.